async-nats/src/jetstream/stream.rs

```diff
@@ -91,6 +94,50 @@ impl Stream {
 
         Ok(Consumer::new(T::try_from_consumer_config(info.config)?))
     }
+
+    pub async fn get_or_create_consumer<T: FromConsumer + IntoConsumerConfig>(
+        &self,
+        name: &str,
+        config: T,
+    ) -> Result<Consumer<T>, Error> {
+        let subject = format!(
```

Comment:
until we have proper status codes and errors it's easier to send request manually here instead of using `consumer.info()`.

Reply:
Noted, will wrap up the #474 now.